test/python/test_reordering.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name,no-value-for-parameter,broad-except
+
+# Copyright 2018 IBM RESEARCH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+"""Tests for bit reordering fix."""
+
+import unittest
+import qiskit as qk
+from qiskit.wrapper import register, available_backends, get_backend, execute
+from .common import requires_qe_access, QiskitTestCase
+
+
+def lowest_pending_jobs(list_of_backends):
+    """Returns the backend with lowest pending jobs."""
+    by_pending_jobs = sorted(list_of_backends,
+                             key=lambda x: get_backend(x).status['pending_jobs'])
+    return by_pending_jobs[0]
+
+
+@requires_qe_access
+def _authenticate(QE_TOKEN, QE_URL):
+    sim_backend = 'local_qiskit_simulator'
+    try:
+        register(QE_TOKEN, QE_URL)
+        real_backends = available_backends({'simulator': False})
+        real_backend = lowest_pending_jobs(real_backends)
+    except Exception:
+        real_backend = None
+
+    return sim_backend, real_backend
+
+
+sim, real = _authenticate()
+
+
+@unittest.skipIf(not real, 'no remote device available.')
+class TestBitReordering(QiskitTestCase):
```

Comment:
Can you revise these tests? They seem to be raising an Exception intermitently:

```
qiskit/wrapper/_wrapper.py:150: in execute
    wait, timeout, skip_translation)
qiskit/_compiler.py:77: in execute
    result = backend.run(q_job)
qiskit/backends/ibmq/ibmqbackend.py:125: in run
    _reorder_bits(this_result)  # TODO: remove this after Qobj

    def _reorder_bits(result):
        """temporary fix for ibmq backends.
        for every ran circuit, get reordering information from qobj
        and apply reordering on result"""
        for idx, circ in enumerate(result._qobj['circuits']):
    
            # device_qubit -> device_clbit (how it should have been)
            measure_dict = {op['qubits'][0]: op['clbits'][0]
                            for op in circ['compiled_circuit']['operations']
                            if op['name'] == 'measure'}
    
            res = result._result['result'][idx]
            counts_dict_new = {}
>           for item in res['data']['counts'].items():
E           TypeError: string indices must be integers
```

when running against `ibmqx4` at least. It seems it might be related to a timeout which is not handled properly during `run()` - the contents of `result._result` on one of those failures where:
```
{'backend': 'ibmqx4',
 'job_id': '(redacted)',
 'name': '3ZwDQis6FwX0j5gRpYvmcAEZ273upm',
 'result': 'QISkit Time Out',
 'status': 'ERROR'}
```

From a practical point of view, they are also a bit problematic: since they depend on a real device, the running time of the tests can be rather large (in my tests, they ranged from 150 to 170 seconds, which is rather large considering that the [current running time for all the test suite is ~120 seconds](https://travis-ci.org/QISKit/qiskit-sdk-py/jobs/373862106#L1534). I'd really like to avoid that one way or the other, by making them opt-in or similar - any ideas?